tests/phpunit/tests/wp-token-map/wpTokenMap.php

```diff
@@ -0,0 +1,377 @@
+<?php
+/**
+ * Unit tests covering WP_Token_Map functionality.
+ *
+ * @package WordPress
+ *
+ * @since 6.6.0
+ *
+ * @coversDefaultClass WP_Token_map
+ */
+class Tests_WpTokenMap extends WP_UnitTestCase {
+	/**
+	 * Returns an associative array whose keys are tokens to replace and
+	 * whose values are the replacement strings for those tokens.
+	 *
+	 * This function is here to help avoid bloating this specific test file.
+	 * For example, the HTML5 dataset is very large and best served as a
+	 * separate file.
+	 *
+	 * @ticket 60698.
+	 *
+	 * @param string $dataset_name Which dataset to return.
+	 * @return array The dataset as an associative array.
+	 */
+	private static function get_test_input_array( $dataset_name ) {
+		static $html5_character_references = null;
+
+		switch ( $dataset_name ) {
+			case 'ANIMALS':
+				return self::ANIMAL_EMOJI;
+
+			case 'HTML5':
+				if ( ! isset( $html5_character_references ) ) {
+					$dataset = json_decode( file_get_contents( __DIR__ . '/../../data/html5-entities.json' ), JSON_OBJECT_AS_ARRAY ); // phpcs:ignore.
```

Comment:
used it! makes the testing a bit more annoying locally because I had to polyfill that function in my `bootstrap.php`, but I like the built-in error-checking in the helper.